### DISCUSSION TITLE
Why is torch a dependency?

### DISCUSSION THREAD

**Discussion by chemwolf6922** [open-webui/open-webui#5820] [General]
**Is your feature request related to a problem? Please describe.**
![image](https://github.com/user-attachments/assets/78eba5ba-705d-45b3-98a5-f24829746af3)
This is the only place where torch is used. I do not see why this is better than calculating with numpy. 
Torch is a gigantic package and this is using it like a calculator. This is a great waste of space in the docker image. 

**Describe the solution you'd like**
I suggest replace torch with numpy and remove it as a dependency. 

**Describe alternatives you've considered**

**Additional context**
The docker image size keeps growing. 5.7GB is not a reasonable size for a "UI".

**Comment by tjbck** [comment]
Take a deep look at Provides-Extra section here: https://pypi.org/project/transformers/ 😃

https://github.com/ollama-webui/ollama-webui-lite would be a good alternative in case anyone's looking for just the "UI", which Open WebUI is not.
